contracts/rewarders/SimpleRewarderPerSec.sol

```diff
@@ -119,107 +141,134 @@ contract SimpleRewarderPerSec is IRewarder, BoringOwnable, ReentrancyGuard {
             uint256 lpSupply = lpToken.balanceOf(address(MCA));
 
             if (lpSupply > 0) {
-                uint256 timeElapsed = block.timestamp.sub(pool.lastRewardTimestamp);
-                uint256 tokenReward = timeElapsed.mul(tokenPerSec);
-                pool.accTokenPerShare = pool.accTokenPerShare.add((tokenReward.mul(ACC_TOKEN_PRECISION) / lpSupply));
+                for (uint i = 0; i < 4; i++){
+                    uint256 timeElapsed = block.timestamp.sub(pool.lastRewardTimestamp);
+                    uint256 tokenReward = timeElapsed.mul(tokenPerSec[i]);
+                    pool.accTokenPerShare[i] = pool.accTokenPerShare[i].add((tokenReward.mul(ACC_TOKEN_PRECISION) / lpSupply));
+                }
             }
-
             pool.lastRewardTimestamp = block.timestamp;
             poolInfo = pool;
         }
     }
 
     /// @dev Sets the distribution reward rate. This will also update the poolInfo.
     /// @param _tokenPerSec The number of tokens to distribute per second
-    function setRewardRate(uint256 _tokenPerSec) external onlyOwner {
+    function setRewardRate(uint256[] memory _tokenPerSec) external onlyOwner {
         updatePool();
+        uint256[] memory oldRate = new uint256[](4);
 
-        uint256 oldRate = tokenPerSec;
-        tokenPerSec = _tokenPerSec;
+        for (uint i = 0; i < 4; i++){
+            uint256 oldRates = tokenPerSec[i]; 
 
-        emit RewardRateUpdated(oldRate, _tokenPerSec);
+            oldRate[i] = oldRates;
+            tokenPerSec[i] = _tokenPerSec[i];
+        }
+        emit RewardRateUpdated(oldRate, _tokenPerSec); 
     }
 
     /// @dev Function called by MasterChefAxial whenever staker claims AXIAL harvest. Allows staker to also receive a 2nd reward token.
     /// @param _user Address of user
     /// @param _lpAmount Number of LP tokens the user has
     function onAxialReward(address _user, uint256 _lpAmount) external override onlyMCA nonReentrant {
+        console.log("Inside onAxialReward");
+        address[4] memory rewardTokens = rewardTokenArray(rewardToken1, rewardToken2, rewardToken3, rewardToken4); 
```

Comment:
This line of code here seems useless, becuase https://github.com/Snowball-Finance/axial-protocol/pull/8/files#diff-62c1fae81585dcf5a7a54891c7d3a925cebe3cd59fb85bf068424253cd2fe8e7R131 does seems to be returning anything, unless am totally missing solidity syntax here.